tools/src/tensil/tools/compiler/Estimator.scala

```diff
@@ -0,0 +1,128 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright © 2019-2022 Tensil AI Company */
+
+package tensil.tools.compiler
+
+import tensil.Architecture
+
+class Estimate(initCycles: Long, initEnergy: Long) {
+  private var curCycles = initCycles
+  private var curEnergy = initEnergy
+
+  def cycles = curCycles
+  def energy = curEnergy
+
+  def splitCycles(cyclesToSplit: Long): Estimate = {
+    require(cyclesToSplit < cycles)
+
+    val energyToSplit =
+      ((cyclesToSplit.toFloat / cycles.toFloat) * energy.toFloat).toLong
+
+    curCycles -= cyclesToSplit
+    curEnergy -= energyToSplit
+
+    new Estimate(cyclesToSplit, energyToSplit)
+  }
+}
+
+class Estimator(arch: Architecture) {
+  private var previousOpcode = Opcode.Wait
+  private var previousFlags  = 0
+
+  private val InternalTransferEnergy = 10
+  private val InternalTransferCycles = 1
+
+  private val DRAMTransferEnergy = 100
+
+  private val DRAMTransferSetupReadCycles  = 32
+  private val DRAMTransferSetupWriteCycles = 0
+  private val DRAMTransferReadCycles       = 2
+  private val DRAMTransferWriteCycles      = 1
+  private val DRAMTransferWidthBits        = 128
```

Comment:
I forget, did we say we were going to fold this param into arch? or have a separate "arch" object just for board parameters. in either case no change needed here, just wondering

Reply:
These constants are not used right now, but I wanted to keep them around for the reference.

Reply:
I commented out unused constants.